lib/src/mocks/http.dart

```diff
@@ -36,8 +37,27 @@ void cancelMockRequest(MockBaseRequest request) {
 }
 
 handleMockRequest(MockBaseRequest request) {
-  Iterable matchingExpectations = _expectations
-      .where((e) => e.method == request.method && e.uri == request.uri);
+  Iterable matchingExpectations = _expectations.where((e) {
+    bool methodMatches = e.method == request.method;
+    bool uriMatches = e.uri == request.uri;
+    bool headersMatch;
+    if (e.headers == null) {
+      // Ignore headers check if expectation didn't specify.
+      headersMatch = true;
+    } else if (e.headers.isEmpty) {
+      headersMatch = request.headers.isEmpty;
+    } else {
+      headersMatch = true;
+      e.headers.forEach((header, value) {
```

Comment:
Yeah that's what I wanted because some default headers get added (`content-type`, `content-length`). If the tester wants to verify those as well, they can be including them in the expectation, but I thought it would be annoying to have to include headers that aren't explicitly set